src/scene_battle_rpg2k.cpp

```diff
@@ -953,6 +953,22 @@ bool Scene_Battle_Rpg2k::ProcessActionResults(Game_BattleAlgorithm::AlgorithmBas
 		return ProcessNextAction(BattleActionState_Death, action);
 	}
 
+	// Check if dead enemies get revived to make them visible again
+	if (battle_action_substate == eConditions
+			&& action->IsRevived())
+	{
+		if (action->GetTarget() != nullptr) {
+			auto* target = action->GetTarget();
+			if (target->GetType() == Game_Battler::Type_Enemy) {
+				auto* target_sprite = Game_Battle::GetSpriteset().FindBattler(target);
+				if (target_sprite) {
+					target_sprite->SetForcedAlive(true);
```

Comment:
This "forced alive" thing was an ugly hack to allow the 2k battle system to "kill" the monster and animate it's death later.

Do we need to set this flag here or is `DetectStateChange` enough?

Reply:
Using `DetectStateChange` is indeed sufficient. I have updated this PR now.